invoker/buildpack.go

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2019 The original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package invoker
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/buildpack/libbuildpack/buildplan"
+	"github.com/cloudfoundry/libcfbuildpack/build"
+	"github.com/cloudfoundry/libcfbuildpack/detect"
+	"github.com/projectriff/riff-buildpack/metadata"
+)
+
+const (
+	Error_Initialize          = 101
+	Error_ReadMetadata        = 102
+	Error_DetectedNone        = 103
+	Error_DetectAmbiguity     = 104
+	Error_UnsupportedLanguage = 105
+	Error_DetectInternalError = 106
+)
+
+type Buildpack interface {
+	Name() string
```

Comment:
Maybe rename (or add) to `Key` or something, as this is something that is tested for strict equality with `==` and is not meant to be pretty displayable (like a product name)

Reply:
Renamed to `Id()`